src/caffe/data_transformer.cpp

```diff
@@ -165,8 +165,8 @@ void DataTransformer<Dtype>::Transform(const vector<Datum> & datum_vector,
   const int width = transformed_blob->width();
 
   CHECK_GT(datum_num, 0) << "There is no datum to add";
-  CHECK_LE(datum_num, num) <<
-    "The size of datum_vector must be smaller than transformed_blob->num()";
+  CHECK_EQ(datum_num, num) <<
```

Comment:
what's the point to transform datum/mat into a partially empty blob? I think it is misleading, the data is always added starting from `iter == 0`.
Furthermore, `memory_data_layer` always wants `added_data` is multiple of `batch_size` and `!= 0`, so the datum vector or the mat vector needs to have the same size as `added_data`